test/browser/column/resizable.test.tsx

```diff
@@ -240,3 +240,26 @@ test('should remeasure flex columns when resizing a column', async () => {
   await expect.element(grid).toHaveStyle({ gridTemplateColumns: '79.1406px 919.422px 919.438px' });
   expect(onColumnResize).toHaveBeenCalledOnce();
 });
+
+test('should use columnWidths and onColumnWidthsChange props when provided', async () => {
+  const onColumnWidthsChange = vi.fn();
+  setup<Row, unknown>({
+    columns,
+    rows: [],
+    columnWidths: new Map([
+      ['col1', { width: 101, type: 'measured' }],
+      ['col2', { width: 201, type: 'measured' }]
+    ]),
+    onColumnWidthsChange
+  });
+  const grid = getGrid();
+  await expect.element(grid).toHaveStyle({ gridTemplateColumns: '101px 201px' });
+  const [, col2] = getHeaderCells();
+  await autoResize(col2);
+  expect(onColumnWidthsChange).toHaveBeenCalledExactlyOnceWith(
+    new Map([
+      ['col1', { width: 101, type: 'measured' }],
+      ['col2', { width: 100, type: 'resized' }]
+    ])
+  );
+});
```

Comment:
Can we add a test that tests that we sync the internal state with the external state when the external state changes. We can try to resize after an external change, to make sure the external change remains even while resizing

Reply:
Done